src/Stack.cpp

```diff
@@ -0,0 +1,78 @@
+#include "Stack.h"
+
+Stack::Stack(const size_t& initial_size, const int& initial_value) : m_data(Vector(initial_size, initial_value)) {}
+
+void Stack::push(int value) {
+    m_data.push_back(value);
+}
+
+void Stack::pop() {
+    m_data.pop_back();
+}
+
+int &Stack::top() {
+    return m_data[m_data.size() - 1];
+}
+
+const int &Stack::top() const {
+    return m_data[m_data.size() - 1];
+}
+
+bool Stack::empty() const {
+    return m_data.size() == 0;
+}
+
+size_t Stack::size() const {
+    return m_data.size();
+}
+
+bool Stack::operator==(const Stack &other) const {
+    return m_data.size() == other.m_data.size();
```

Comment:
Why do you check `other`'s size via the private member `m_data` and not with the public method `size()` you implemented?